include/onnxruntime/core/framework/allocator.h

```diff
@@ -144,13 +150,14 @@ class IAllocator {
      @param stream Which stream instance allocated chunk will be used with.
      @param wait_fn If the allocator want to dynamic reuse a chunk from another stream, use this wait_fn to sync on
                     the target stream to make the reuse safe.
-     @returns std::unique_ptr with allocated memory and deleter.
+     @returns std::unique_ptr with allocated memory and deleter. Throws if it cannot allocate memory.
```

Comment:
> Throws if it cannot allocate memory.

Does this mean that IAllocator::Alloc is expected to throw and not return nullptr on error? Looks like the CPU and CUDA EP allocators already do that anyway.

Could document that assumption or enforce that `p != nullptr || count_or_bytes == 0`.